core/src/impl/Kokkos_Core.cpp

```diff
@@ -351,8 +351,24 @@ std::vector<int> Kokkos::Impl::get_visible_devices(
 int Kokkos::Impl::get_gpu(const InitializationSettings& settings) {
   std::vector<int> visible_devices =
       get_visible_devices(settings, get_device_count());
+  int const num_devices = visible_devices.size();
   // device_id is provided
   if (settings.has_device_id()) {
+    int const id = settings.get_device_id();
+    if (id < 0) {
+      std::stringstream ss;
+      ss << "Error: Requested GPU with invalid id '" << id << "'."
+         << " Device id cannot be negative!"
+         << " Raised by Kokkos::initialize(int argc, char* argv[]).\n";
```

Comment:
It could be any overload, right?
```suggestion
         << " Raised by Kokkos::initialize().\n";
```

Reply:
I tend to agree with you but let us fix all occurrences in a follow-up instead.